addons/delivery/models/stock_picking.py

```diff
@@ -122,29 +100,31 @@ def action_done(self):
         return res
 
     @api.multi
-    def put_in_pack(self):
-        res = super(StockPicking, self).put_in_pack()
-        if isinstance(res, dict) and res.get('type'):
-            return res
-        if self.carrier_id and self.carrier_id.delivery_type not in ['base_on_rule', 'fixed']:
-            view_id = self.env.ref('delivery.choose_delivery_package_view_form').id
-            return {
-                'name': _('Package Details'),
-                'type': 'ir.actions.act_window',
-                'view_mode': 'form',
-                'res_model': 'choose.delivery.package',
-                'view_id': view_id,
-                'views': [(view_id, 'form')],
-                'target': 'new',
-                'context': dict(
-                    self.env.context,
-                    current_package_carrier_type=self.carrier_id.delivery_type,
-                    default_stock_quant_package_id=res.id
-                ),
-            }
+    def _pre_put_in_pack_hook(self):
+        res = super(StockPicking, self)._pre_put_in_pack_hook()
+        if not res:
+            if self.carrier_id:
+                return self._set_delivery_packaging()
         else:
             return res
 
+    def _set_delivery_packaging(self):
```

Comment:
ensure_one
+ docstring explaining this method returns an action allowing to set the product packaging on the stock quant package and the shipping weight